src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -342,115 +322,30 @@ public SQLDatasetProducer getProducer(SQLPullRequest pullRequest, PullCapability
 
   @Override
   public SQLWriteResult write(SQLWriteRequest writeRequest) {
-    try {
-      return writeInternal(writeRequest);
-    } catch (InterruptedException e) {
-      LOG.error("Interrupted exception during BigQuery write operation.", e);
-    } catch (BigQueryException bqe) {
-      LOG.error("BigQuery exception during BigQuery write operation", bqe);
-    }
-
-    // Return as a failure if the operation threw an exception.
-    return SQLWriteResult.faiure(writeRequest.getDatasetName());
-  }
-
-  private SQLWriteResult writeInternal(SQLWriteRequest writeRequest) throws BigQueryException, InterruptedException {
     String datasetName = writeRequest.getDatasetName();
-    if (!getClass().getName().equals(writeRequest.getOutput().getSqlEngineClassName())) {
-      LOG.debug("Got output for another SQL engine {}, skipping", writeRequest.getOutput().getSqlEngineClassName());
-      return SQLWriteResult.unsupported(datasetName);
-    }
 
-    Map<String, String> arguments = writeRequest.getOutput().getArguments();
-
-    String jobId = arguments.get(SQL_OUTPUT_JOB_ID);
-    String sourceTableName = datasets.get(writeRequest.getDatasetName()).getBigQueryTableName();
-    BigQuerySinkConfig sinkConfig = GSON.fromJson(arguments.get(SQL_OUTPUT_CONFIG), BigQuerySinkConfig.class);
-    Schema schema = GSON.fromJson(arguments.get(SQL_OUTPUT_SCHEMA), Schema.class);
-    String destinationDataset = sinkConfig.getDataset();
-    String destinationTableName = arguments.get(SQL_OUTPUT_TABLE);
-    String destinationProject = sinkConfig.getDatasetProject();
-    List<String> fields = GSON.fromJson(arguments.get(SQL_OUTPUT_FIELDS), LIST_OF_STRINGS_TYPE);
-    boolean allowSchemaRelaxation = sinkConfig.isAllowSchemaRelaxation();
-    JobInfo.WriteDisposition writeDisposition = sinkConfig.getWriteDisposition();
-
-    if (Operation.INSERT != sinkConfig.getOperation()) {
-      LOG.warn("Direct table copy is only supported for INSERT operations.");
+    // Check if direct sink write is enabled. If not, skip.
+    if (!sqlEngineConfig.shouldUseDirectSinkWrite()) {
```

Comment:
Should we add a log statement here?

Reply:
Since the user didn't enable this feature, we can simply skip this